### ISSUE TITLE
Train, Valid and Test masks for CitationGraphDatasets return a 0/1 array or tensor

### ISSUE THREAD

**Issue by bala-srini** [dmlc/dgl#1798]
## 🐛 Bug

Train, Valid and Test masks for CitationGraphDatasets to return a 0/1 array or tensor - Can you please change them to boolean - so that it can be used effectively as mask directly

# To Reproduce

1.
1.
1.

## Environment

 - DGL Version (e.g., 1.0):
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3):
 - OS (e.g., Linux):
 - How you installed DGL (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version (if applicable):
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:

**Comment by classicsong** [comment]
Hi, we are reorganizing our Builtin dataset. Please see #1776.
For CitationGraphDatasets, the mask will be bool tensor for PyTorch and TensorFlow backend.

**Comment by classicsong** [comment]
In this PR https://github.com/dmlc/dgl/pull/1902
Citation graph will return an bool tensor when the backend is Torch or TensorFlow. You can refer to GCN or GAT examples for how to use it.
